libs/datamodel/core/tests/attributes/default_composite_negative.rs

```diff
@@ -0,0 +1,411 @@
+}
+
+#[test]
+fn must_error_if_now_function_is_used_for_fields_that_are_not_datetime() {
+    let dml = indoc! {r#"
+        type Composite {
+          foo String @default(now())
+        }
+    "#};
+
+    let error = datamodel::parse_schema(dml).map(drop).unwrap_err();
+
+    let expectation = expect![[r#"
+        [1;91merror[0m: [1mError parsing attribute "@default": The function `now()` cannot be used on fields of type `String`.[0m
+          [1;94m-->[0m  [4mschema.prisma:2[0m
+        [1;94m   | [0m
+        [1;94m 1 | [0mtype Composite {
+        [1;94m 2 | [0m  foo String @[1;91mdefault(now())[0m
+        [1;94m   | [0m
+    "#]];
+
+    expectation.assert_eq(&error)
+}
+
+#[test]
+fn must_error_if_autoincrement_function_is_used() {
+    let dml = indoc! {r#"
+        type Composite {
+          foo String @default(autoincrement())
+        }
+    "#};
+
+    let error = datamodel::parse_schema(dml).map(drop).unwrap_err();
+
+    let expectation = expect![[r#"
+        [1;91merror[0m: [1mError parsing attribute "@default": The function `autoincrement()` is not a supported on composite fields.[0m
+          [1;94m-->[0m  [4mschema.prisma:2[0m
+        [1;94m   | [0m
+        [1;94m 1 | [0mtype Composite {
+        [1;94m 2 | [0m  foo String @[1;91mdefault(autoincrement())[0m
+        [1;94m   | [0m
+    "#]];
+
+    expectation.assert_eq(&error)
+}
+
+#[test]
+fn must_error_if_default_value_for_enum_is_not_valid() {
+    let dml = indoc! {r#"
+      type Composite {
+        enum A @default(B)
+      }
+
+      enum A {
+        A
+      }
+  "#};
+
+    let error = datamodel::parse_schema(dml).map(drop).unwrap_err();
+
+    let expectation = expect![[r#"
+        [1;91merror[0m: [1mError parsing attribute "@default": The defined default value `B` is not a valid value of the enum specified for the field.[0m
+          [1;94m-->[0m  [4mschema.prisma:2[0m
+        [1;94m   | [0m
+        [1;94m 1 | [0mtype Composite {
+        [1;94m 2 | [0m  enum A @[1;91mdefault(B)[0m
+        [1;94m   | [0m
+    "#]];
+
+    expectation.assert_eq(&error)
+}
+
+#[test]
+fn must_error_if_scalar_default_on_unsupported() {
+    let dml = indoc! {r#"
+        datasource db1 {
+          provider = "postgresql"
+          url = "postgresql://"
+        }
+
+        type Composite {
+          balance Unsupported("some random stuff") @default(12)
+        }
+    "#};
+
+    let error = datamodel::parse_schema(dml).map(drop).unwrap_err();
+
+    let expectation = expect![[r#"
+        [1;91merror[0m: [1mError parsing attribute "@default": Composite field of type `Unsupported` cannot have default values.[0m
+          [1;94m-->[0m  [4mschema.prisma:7[0m
+        [1;94m   | [0m
+        [1;94m 6 | [0mtype Composite {
+        [1;94m 7 | [0m  balance Unsupported("some random stuff") @[1;91mdefault(12)[0m
+        [1;94m   | [0m
+    "#]];
+
+    expectation.assert_eq(&error)
+}
```

Comment:
I'm wondering where we stand on `Unsupported(...)` fields on composite types. We should already have tests, but I'm not sure we do.

Reply:
To be figured out and addressed later: https://github.com/prisma/prisma/issues/11282